# Description: let initial directory selection work correctly
# Ubuntu: 
# Upstream: https://bugzilla.gnome.org/557689
#
From b6d9bb6e1e6e916fc2ef25f5729b30ba91979597 Mon Sep 17 00:00:00 2001
From: Cody Russell <bratsche@gnome.org>
Date: Mon, 05 Apr 2010 17:57:33 +0000
Subject: Fix for bug #557689 and bug #402349, where opening a file chooser in

folder browse mode would require the user to click on something
before clicking "Open".
---
diff --git a/gtk/gtkfilechooserdefault.c b/gtk/gtkfilechooserdefault.c
index 81150d0..1d60429 100644
--- a/gtk/gtkfilechooserdefault.c
+++ b/gtk/gtkfilechooserdefault.c
@@ -8594,8 +8594,16 @@ gtk_file_chooser_default_should_respond (GtkFileChooserEmbed *chooser_embed)
       entry = GTK_FILE_CHOOSER_ENTRY (impl->location_entry);
       check_save_entry (impl, &file, &is_well_formed, &is_empty, &is_file_part_empty, &is_folder);
 
-      if (is_empty || !is_well_formed)
-	return FALSE;
+      if (!is_well_formed)
+        return FALSE;
+
+      if (is_empty)
+        {
+          if (impl->action == GTK_FILE_CHOOSER_ACTION_SAVE)
+            return FALSE;
+
+          goto file_list;
+        }
 
       g_assert (file != NULL);
 
--
cgit v0.8.3.1

